docs/proposals/kubelet-systemd.md

```diff
@@ -0,0 +1,277 @@
+
+This proposal asserts that the reservation of compute resources for any associated
+container runtime daemons is tracked by the operator under the system-reserved or
+kubernetes-reserved values at this time and any enforced limits are set by the
+operator specific to the container runtime.
+
+For future consideration, if the `kubelet` attempts to enforce `--kube-reserved`, this
+proposal asserts that the reservation associated with any particular container runtime
+should be split out at that time.  In a `systemd` environment, this proposal asserts that
+you cannot mandate the container runtime daemon is in the same `slice` as the Kubernetes
+components.
+
+**Docker**
+
+If the `kubelet` is configured with the `container-runtime` set to `docker`, the
+`kubelet` will detect the `service` associated with the `docker` daemon and use that
+to do local node accounting.  If an operator wants to impose runtime limits on the
+`docker` daemon to control resource usage, the operator should set those explicitly in
+the `service` unit that launches `docker`.  The `kubelet` will not set any limits itself
+at this time and will assume whatever budget was set aside for `docker` was included in
+either `--kube-reserved` or `--system-reserved` reservations.
+
+**Rocket**
+
+Rocket has no client/server daemon, and therefore has no explicit requirements on container-runtime
+reservation.
+
+### Node allocable
+
+The proposal makes no changes to the definition.
+
+The node will report a set of allocable compute resources defined as follows:
+
+`[Allocatable] = [Node Capacity] - [Kube-Reserved] - [System-Reserved]`
+
+### OOM Score Adjustment
+
+The `kubelet` at bootstrapping will set the `oom_score_adj` value for Kubernetes
+daemons, and any dependent container-runtime daemons.
+
+If `container-runtime` is set to `docker`, then set its `oom_score_adj=-900`
+
+### Linux Kernel Parameters
+
+The `kubelet` will set the following:
+
+* `sysctl -w vm.overcommit_memory=1`
+* `sysctl -w vm.panic_on_oom=0`
+* `sysctl -w vm.overcommit_memory=1`
+* `sysctl -w kernel/panic=1`
+* `sysctl -w kernel/panic_on_oops=10`
+
+## kubelet cgroup-root behavior under systemd
+
+The `kubelet` supports a `cgroup-root` flag which is the optional root `cgroup` to use for pods.
+
+This flag should be treated as a pass-through to the underlying configured container runtime.
+
+For example, if the container runtime is `docker` and its using the `systemd` cgroup driver, then
+it should take the form of a `systemd` slice.  For example, `--cgroup-root=foo-bar.slice` would parent
+all of the pod's container under the `foo-bar.slice` part of the hierarchy.
+
+## kubelet accounting for end-user pods
+
+This proposal re-enforces that it is inappropriate at this time to depend on `--cgroup-root` as the
+primary mechanism to distinguish and account for end-user pod compute resource usage.
+
+Instead, the `kubelet` can and should sum the usage of each running `pod` on the node to account for
```

Comment:
Kubelet has to perform accounting. So +1. 
I'd even clarify that `--cgroup-root` is not recommended for most standalone kubernetes installations.